R/PredictionSurv.R

```diff
@@ -13,12 +13,22 @@
 #' learner = lrn("surv.kaplan")
 #' p = learner$train(task, row_ids = 1:20)$predict(task, row_ids = 21:30)
 #' head(as.data.table(p))
+#' class(p$data$distr) # survival matrix stored
+#' p$distr # Matdist
 PredictionSurv = R6Class("PredictionSurv",
   inherit = Prediction,
   public = list(
     #' @description
     #' Creates a new instance of this [R6][R6::R6Class] class.
     #'
+    #' @details
+    #' Upon initialization, the `distr` input will be coerced to a survival matrix
+    #' or array (accessible via `$data$distr`) if it's a [Distribution][distr6::Distribution]
+    #' object. The active field `$distr` always returns a distribution
```

Comment:
I see in the code:
```
distr = function() {
      if (inherits(self$data$distr, "Distribution")) {
        return(self$data$distr)
      }

      private$.distrify_survarray(self$data$distr %??% NA_real_)
    }
```
And the `distrify_survarray` checks for `matrix` (or now `array`). The only case that this is not true is if upon initialization `distr` was neither a `Distribution`, an `array` or a `matrix`. Do we have that case? You mentioned fully parametic distributions or so?

Reply:
```
 if (inherits(self$data$distr, "Distribution")) {
        return(self$data$distr)
      }
```

Checks if it's a `Distribution`, if it is then it returns the same object (as `Distribution`). So not always coerced to matrix or array

Reply:
Updated doc, it's considerably better now